### PR TITLE
Add missing documentation on span objects 'detail' attribute

### PR DESCRIPTION
**Pull request by gchan**
PIE-1010

The docs are missing attributes for span objects (for HTTP and SQL).

A customer reached out about this as they were trying to figure out how
to POST the URL, method, and library details to the span object.

Customer message
https://buildkite-community.slack.com/archives/C03HXN4ARB5/p1660262430972669?thread_ts=1660256140.870549&cid=C03HXN4ARB5

Relevant code
https://github.com/buildkite/buildkite/blob/1e04bd5774c1c54927d15668b35c9bdfeae479f9/app/models/analytics/execution/ingestion_result.rb#L123

Docs page
https://buildkite.com/docs/test-analytics/importing-json#json-test-results-data-reference-test-result-objects

### CODE REVIEW

File: pages/test_analytics/importing_json.md.erb [hunk 1]
```diff
@@ -137,6 +137,7 @@ Schematically, the JSON test results data is like this:
    - [Test result](#json-test-results-data-reference-test-result-objects)
       * [History](#json-test-results-data-reference-history-objects)
          - [Spans](#json-test-results-data-reference-span-objects)
+             * [Detail](#json-test-results-data-reference-detail-objects)
```
Comment (gchan) [inline pages/test_analytics/importing_json.md.erb:140]: This doesn't seem to render correctly. Any thoughts?

Screenshot from 2022-08-15 09-31-11

File: data/test_analytics_json_fields_detail.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+fields:
+- name: method
+  required: false
```
Comment (gchan) [inline data/test_analytics_json_fields_detail.yaml:3]: I'm not sure if we can do conditional 'required' attributes. 

I've noted in the 'desc' attribute that the attribute is required if the parent object has a specific attribute.

**Comment by gchan** [comment]
@plaindocs @karensawrey 
I've left this in draft due to an unresolved issue, but it is ready for an initial review.

File: pages/test_analytics/importing_json.md.erb (hunk 1 above)
Comment (karensawrey) [reply pages/test_analytics/importing_json.md.erb:140]: They
    render
          according
               to your indentation :)

Comment (karensawrey) [inline pages/test_analytics/importing_json.md.erb:140]:
```suggestion
   - [Test result](#json-test-results-data-reference-test-result-objects)
      * [History](#json-test-results-data-reference-history-objects)
   - [Spans](#json-test-results-data-reference-span-objects)
      * [Detail](#json-test-results-data-reference-detail-objects)
```

Comment (karensawrey) [reply pages/test_analytics/importing_json.md.erb:140]: This should fix the outright weird behavior of the item if that's the levels you want in your list.

File: pages/test_analytics/importing_json.md.erb [hunk 3]
```diff
@@ -315,5 +316,56 @@ It represents, for example, the duration of an individual database query within
...
   "duration": 0.0006910000229254365
+  "detail": {
+    ...
+  }
+}
+```
+
+### Detail objects
+
+A detail object contains additional information about the span.
```
Comment (karensawrey) [inline pages/test_analytics/importing_json.md.erb:327]:
```suggestion
Detail objects contains additional information about the span.
```

**Comment by karensawrey** [comment]
Added my 2 cents for now.

File: pages/test_analytics/importing_json.md.erb (hunk 1 above)
Comment (gchan) [reply pages/test_analytics/importing_json.md.erb:140]: I want 'spans' to be under history, and 'detail' under 'span'. Seems like 5 levels of nesting is not allowed/working.
If I remove the top level bullet point, it works.

Screenshot from 2022-08-15 13-22-38

Comment (gchan) [reply pages/test_analytics/importing_json.md.erb:140]: Seems like 5 levels is not working. Removing the top level "a test result array of" solves the problem.

File: pages/test_analytics/importing_json.md.erb (hunk 3 above)
Comment (gchan) [reply pages/test_analytics/importing_json.md.erb]: Resolved

**Comment by gchan** [comment]
I've resolved the nesting problem so I've marked the PR as ready for review

File: data/test_analytics_json_fields_detail.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    - POST
+    - PUT
+    - DELETE
+    - PATCH
+    - HEAD
+    - CONNECT
+    - OPTIONS
+    - TRACE
+  desc: |
+      A HTTP request method (required if parent span has a 'section' of 'http')
```
Comment (plaindocs) [inline data/test_analytics_json_fields_detail.yaml:16]:
```suggestion
      A HTTP request method (required if parent span has a `section` of `http`)
```

File: data/test_analytics_json_fields_detail.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  examples:
+- name: url
+  required: false
+  type: string
+  desc: |
+      The URL requested (required if parent span has a 'section' of 'http')
```
Comment (plaindocs) [inline data/test_analytics_json_fields_detail.yaml:22]:
```suggestion
      The URL requested (required if parent span has a `section` of `http`)
```

File: data/test_analytics_json_fields_detail.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  examples:
+- name: lib
+  required: false
+  type: string
+  desc: |
+     The library used to make the HTTP request (required if parent span has a 'section' of 'http')
```
Comment (plaindocs) [inline data/test_analytics_json_fields_detail.yaml:28]:
```suggestion
     The library used to make the HTTP request (required if parent span has a `section` of `http`)
```

File: data/test_analytics_json_fields_detail.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  examples:
+- name: query
+  required: false
+  type: string
+  desc: |
+     The SQL query (required if parent span has a 'section' of 'sql')
```
Comment (plaindocs) [inline data/test_analytics_json_fields_detail.yaml:34]:
```suggestion
     The SQL query (required if parent span has a `section` of `sql`)
```

File: data/test_analytics_json_fields_span.yaml [hunk 8]
```diff
@@ -28,3 +28,9 @@ fields:
   desc: |
       How long the span took to run
   examples:
+- name: detail
+  required : false
+  desc: |
+      A detail object containing further details on the span. Required when 'section' is 'http', or 'sql'.
```
Comment (plaindocs) [inline data/test_analytics_json_fields_span.yaml:34]:
```suggestion
      A detail object containing further details on the span. Required when `section` is `http`, or `sql`.
```

**Review by plaindocs — APPROVED** [review]
This is great, thanks for jumping in @gchan !!

**Comment by plaindocs** [comment]
@olyism any idea about this wrapping? 

image

**Comment by gchan** [comment]
Would love to hit merge on this @olyism  if you're happy with the current state!

**Comment by olyism** [comment]
@gchan sorry took me a while to get back. The table wrap isn't perfect but I'm happy for the Docs changes to go live so we're not holding up the content changes.

> @olyism any idea about this wrapping?

That wrapping is tricky, as we either wrap, or nowrap and break the table layout. I've got a backlog ticket to address tables holistically. I think something like a horizontal overflow scroll might work but this wouldn't be in the scope of this PR.

**Pull request merged**